mava/networks/gnn.py

```diff
@@ -0,0 +1,234 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence
+
+import chex
+import jraph
+import jraph._src.models as jraph_models
+import jraph._src.utils as jraph_utils
+from flax import linen as nn
+from jax import numpy as jnp
+from jraph import GraphsTuple as JraphGraphsTuple
+
+from mava.networks.torsos import MLPTorso, _parse_activation_fn
+from mava.types import Observation
+from mava.utils.jraph_utils import batched_graph_to_single_graph
+
+
+class GNN(nn.Module):
+    """A parent class for all GNN models.
+    This is used so that we can identify GNN models in base actor and critic networks.
+    """
+
+    pass
+
+
+class InforMARLNbrhdAggregationTorso(GNN):
+    """InforMARL Actor Network.
+    For more details see: https://arxiv.org/abs/2211.02127
+    """
+
+    attention_query_layer_sizes: Sequence[int]
+    use_layer_norm: bool
+    activation: str
+
+    num_heads: int
+    num_attention_layers: int
+
+    @nn.compact
+    def __call__(self, observation: Observation) -> chex.Array:
+        assert observation.graph is not None, "Graph is not provided in the observation"
+
+        obs = observation.agents_view
+        graph = observation.graph
+        assert graph.nodes is not None, "There are no node features in the graph"
+
+        num_timesteps, num_envs, num_agents, num_nodes_per_graph, *_ = graph.nodes.shape
```

Comment:
nit: a convention in mava is to use single capital letters for common dimensions for example [here](https://github.com/instadeepai/Mava/blob/develop/mava/networks/utils/sable/decode.py#L26-L30). It would be great to make one of these comments at the top of this file as I see you also do this shape unpacking below.